src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -281,35 +284,51 @@ public ResponseEntity<StudentExam> submitStudentExam(@PathVariable Long courseId
     public ResponseEntity<StudentExam> getStudentExamForConduction(@PathVariable Long courseId, @PathVariable Long examId, @PathVariable Long studentExamId,
             HttpServletRequest request) {
         long start = System.currentTimeMillis();
-        User user = userRepository.getUserWithGroupsAndAuthorities();
-        log.debug("REST request to get the student exam of user {} for exam {}", user.getLogin(), examId);
+        User currentUser = userRepository.getUserWithGroupsAndAuthorities();
+        log.debug("REST request to get the student exam of user {} for exam {}", currentUser.getLogin(), examId);
 
         StudentExam studentExam = studentExamRepository.findByIdWithExercisesElseThrow(studentExamId);
 
-        if (!user.equals(studentExam.getUser())) {
+        if (!currentUser.equals(studentExam.getUser())) {
             throw new AccessForbiddenException("Current user is not the user of the requested student exam");
         }
-        studentExamAccessService.checkCourseAndExamAccessElseThrow(courseId, examId, user, studentExam.isTestRun(), false);
+
+        if (studentExam.isTestRun()) {
+            // this check is quite expensive, so we only so it for test runs
+            studentExamAccessService.checkCourseAndExamAccessElseThrow(courseId, examId, currentUser, true, false);
+        }
+        else {
+            // those checks are good enough and less expensive, because they do not involve additional database queries
+            checkExamConfigElseThrow(studentExam, examId, courseId);
+        }
```

Comment:
Not really. `checkExamConfigElseThrow` basically checks that the client input for `examId` and `courseId` are correctly matching the student exam which is fetched from the DB. While this is not 100% needed, because we could construct the whole REST URL differently (i.e. without `examId` and `courseId`), we should check it for the current implementation. Our web client would not use different values here, but a random user could do that by sending respective REST Calls directly to the server.